source/imaer-util/src/main/java/nl/overheid/aerius/validation/BuildingValidator.java

```diff
@@ -31,42 +33,58 @@ public final class BuildingValidator {
-        errors.add(new AeriusException(ImaerExceptionReason.BUILDING_HEIGHT_TOO_LOW, building.getLabel()));
-      } else if (building.getHeight() == 0) {
-        warnings.add(new AeriusException(ImaerExceptionReason.BUILDING_HEIGHT_ZERO, building.getLabel()));
-      }
+  private static void checkBuildingGeometry(final BuildingFeature feature, final BuildingLimits buildingLimits,
+      final List<AeriusException> errors) {
+    // Only polygon and point geometries are supported.
+    if (!(feature.getGeometry() instanceof Polygon
+        || (buildingLimits.isCircularBuildingSupported() && feature.getGeometry() instanceof Point))) {
+      errors.add(new AeriusException(ImaerExceptionReason.GML_GEOMETRY_NOT_PERMITTED, feature.getProperties().getLabel()));
     }
   }
 
-  private static void checkBuildingDiameter(final List<BuildingFeature> buildings, final List<AeriusException> errors) {
-    for (final BuildingFeature feature : buildings) {
-      final Building building = feature.getProperties();
-      // When the geometry is a point, that indicates that a circular building is defined.
-      // A circular building consists of a point and a positive diameter.
-      if (feature.getGeometry() instanceof Point && building.getDiameter() <= 0) {
-        errors.add(new AeriusException(ImaerExceptionReason.CIRCULAR_BUILDING_INCORRECT_DIAMETER, building.getLabel()));
-      }
+  private static void checkBuildingHeight(final BuildingFeature feature, final BuildingLimits buildingLimits,
+      final List<AeriusException> errors, final List<AeriusException> warnings) {
+    final Building building = feature.getProperties();
+    if (building.getHeight() < buildingLimits.buildingHeightMinimum()) {
+      errors.add(new AeriusException(ImaerExceptionReason.BUILDING_HEIGHT_TOO_LOW, building.getLabel()));
+    } else if (building.getHeight() == 0) {
+      // This warning only comes into play when the normal building height minimum is 0 or less.
+      // Some models have a higher minimum height, and in that case we don't want this warning to pop up.
+      // Other models do allow it, but modeling it that way would be a bit weird, hence the warning.
+      warnings.add(new AeriusException(ImaerExceptionReason.BUILDING_HEIGHT_ZERO, building.getLabel()));
+    } else if (building.getHeight() > buildingLimits.buildingHeightMaximum()) {
+      errors.add(new AeriusException(ImaerExceptionReason.BUILDING_HEIGHT_TOO_HIGH, building.getLabel()));
+    }
+  }
+
+  private static void checkBuildingDiameter(final BuildingFeature feature, final BuildingLimits buildingLimits,
+      final List<AeriusException> errors) {
+    final Building building = feature.getProperties();
+    // When the geometry is a point, that indicates that a circular building is defined.
+    // A circular building consists of a point and a positive diameter.
+    final double diameter = building.getDiameter();
+    if (feature.getGeometry() instanceof Point
```

Comment:
I would suggest to also add the support for circular building here
```suggestion
    if (buildingLimits.isCircularBuildingSupported() && feature.getGeometry() instanceof Point
```